Update cooperative cancellation sample (#6417)
Particular/docs.particular.net: samples/cooperative-cancellation/sample.md

---
title: Cooperative cancellation
summary: Stopping both the endpoint and a long running handler with a cancellation token
reviewed: 2024-01-20
component: Core
---

Run the solution. A single console application starts up: `Server`.

## Code walk-through

> When the endpoint is started, a message is sent to the endpoint, which triggers a long-running message handler.  The handler enters an infinite loop, logging a message every two seconds, by calling the async `Task.Delay()` operation.  The `CancellationToken`, that is available in the context object where messages are being handled, is passed to `Task.Delay()` to cancel the delay operation if the `context.CancellationToken.IsCancellationRequested` property is set to `true`.

Reviewer: ```suggestion
When the endpoint is started, a message is sent to the endpoint, which triggers a long-running message handler.  The handler enters an infinite loop, logging a message every two seconds, by calling the async `Task.Delay()` operation.  The `CancellationToken` provided by the message handling context object is passed to `Task.Delay()` to cancel the delay operation if the `context.CancellationToken.IsCancellationRequested` property is set to `true`.
```

---
title: Cooperative cancellation
summary: Stopping both the endpoint and a long running handler with a cancellation token
reviewed: 2024-01-20
component: Core
---

Run the solution. A single console application starts up: `Server`.

## Code walk-through

> When the endpoint is started, a message is sent to the endpoint, which triggers a long-running message handler that enters a loop simulating a task that takes 3 seconds to execute, like writing a bunch of records to a database, by calling the async `Task.Delay()` operation.

Reviewer: Not quite accurate, right? You've made it take infinite time to execute, but just log something every 3 seconds.
